programs/vote/src/vote_state/mod.rs

```diff
@@ -510,12 +718,6 @@ impl VoteState {
             return Err(VoteError::TooManyVotes);
         }
 
-        // check_slots_are_valid()` ensures we don't process any states
-        // that are older than the current state
-        if !self.votes.is_empty() {
-            assert!(new_state.back().unwrap().slot > self.votes.back().unwrap().slot);
-        }
-
```

Comment:
This needs a feature gate

Reply:
This function is already behind a feature gate https://github.com/solana-labs/solana/blob/66fa8f966753b732facbbd66ff5b549be796ae3c/programs/vote/src/vote_instruction.rs#L460-L473, so should cause no difference in behavior